packages/x-data-grid-pro/src/hooks/features/dataSource/gridDataSourceSelector.ts

```diff
@@ -14,15 +15,32 @@ const computeStartEnd = (paginationModel: GridPaginationModel) => {
   return { start, end };
 };
 
+type GridStateProWithRowGrouping = GridStatePro & {
+  rowGrouping?: {
+    model: string[];
+  };
+};
+
+const EMPTY_ARRAY: string[] = [];
+
+const gridRowGroupingModelSelector = (state: GridStateProWithRowGrouping) =>
+  state.rowGrouping?.model ?? EMPTY_ARRAY;
+
+export const gridRowGroupingSanitizedModelSelector = createSelector(
+  gridRowGroupingModelSelector,
+  gridColumnLookupSelector,
+  (model, columnsLookup) => model.filter((field) => !!columnsLookup[field]),
+);
```

Comment:
For my understanding, in which cases is the sanitization necessary?

Reply:
And shouldn't this be a memoized selector, considering that it's building a derived array value?

Reply:
When a field is provided in `rowGroupingModel` that doesn't exist (or got removed from) the column definitions.
It's similar to the one we are using in the actual `rowGrouping` [implementation](https://github.com/mui/mui-x/blob/153bce3cd50df281801ab8a4721d3aa43a919023/packages/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingSelector.ts#L12-L16).

Side topic: I initially thought to move this one in the Pro package and export from `/internals` to reuse but went ahead and duplicated it. Would reuse approach will be better/less confusing?

Reply:
If it makes sense, yes I would avoid duplication.

Reply:
You are right, it should be memoized 👍 
I'll do either of:
1. Move the original selectors to Pro and reuse them in both places.
2. Convert these to memoized 

Please vote if you have a preference:
👍 = 1, 🚀 = 2

Reply:
No preference, do what you think is best.

Reply:
Ended up converting the selector to memoized as bringing every related thing to the lower package (like `GridStatePremium` interface) seems a bit too much.